File: .NET/Microsoft.Recognizers.Text.DataTypes.DataDrivenTests/TestTime.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace Microsoft.Recognizers.Text.DataTypes.TimexExpression.Tests
{
    [TestClass]
    public class TestTime
    {
        [TestMethod]
        public void DataTypes_Time_Constructor()
        {
            var t = new Time(23, 45, 32);
            Assert.AreEqual(23, t.Hour);
            Assert.AreEqual(45, t.Minute);
            Assert.AreEqual(32, t.Second);
        }

        [TestMethod]
        public void DataTypes_Time_GetTime()
        {
            var t = new Time(23, 45, 32);
            Assert.AreEqual(85532000, t.GetTime());
        }
    }
}

File: Python/tests/datatypes/test_time.py
from datatypes_timex_expression import Time


def test_datatypes_time_constructor():
    time = Time(23, 45, 32)

    assert time.hour is 23
    assert time.minute is 45
    assert time.second is 32


def test_datatypes_time_gettime():
    time = Time(23, 45, 32)

    assert time.get_time() == 85532000
